controller/src/server.rs

```diff
@@ -0,0 +1,71 @@
+use tonic::{transport::Server, Request, Response, Status};
+
+use scheduler::scheduling_service_server::{SchedulingService, SchedulingServiceServer};
+use scheduler::{SchedulingRequest, SchedulingResponse};
+use axum::{response::Html, routing::get, Router};
+use std::net::SocketAddr;
+use tokio::task;
+
+pub mod scheduler {
+    tonic::include_proto!("orkascheduler"); // The string specified here must match the proto package name
+}
+
+#[derive(Debug, Default)]
+pub struct MySchedulingService {}
+
+#[tonic::async_trait]
+impl SchedulingService for MySchedulingService {
+    async fn schedule(
+        &self,
+        request: Request<SchedulingRequest>,
+    ) -> Result<Response<SchedulingResponse>, Status> {
+        println!("Got a request: {:?}", request);
+
+        let response = SchedulingResponse {
+            status_code: 0,
+            rejection_reason: Some(1),
+        };
+
+        Ok(Response::new(response))
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // GRPC
+    let grpc_addr = "[::1]:50051".parse()?;
+    let scheduler = MySchedulingService::default();
+
+    // Spawn the gRPC server as a tokio task
+    let grpc_thread = task::spawn(async move {
+        println!("Running grpc here: {}", grpc_addr);
```

Comment:
I'm ok with `println` for now, but you should use a more flexible and robust logging mechanism. See e.g. https://docs.rs/log/latest/log/